libkbfs/key_bundle_cache_measured.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	metrics "github.com/rcrowley/go-metrics"
+)
+
+// KeyBundleCacheMeasured delegates to another KeyBundleCache instance but
+// also keeps track of stats.
+type KeyBundleCacheMeasured struct {
+	delegate                  KeyBundleCache
+	getReaderBundleTimer      metrics.Timer
+	getWriterBundleTimer      metrics.Timer
+	putReaderBundleTimer      metrics.Timer
+	putWriterBundleTimer      metrics.Timer
+	hitReaderBundleCountMeter metrics.Meter
+	hitWriterBundleCountMeter metrics.Meter
+}
+
+var _ KeyBundleCache = KeyBundleCacheMeasured{}
+
+// NewKeyBundleCacheMeasured creates and returns a new KeyBundleCacheMeasured
+// instance with the given delegate and registry.
+func NewKeyBundleCacheMeasured(delegate KeyBundleCache, r metrics.Registry) KeyBundleCacheMeasured {
+	getReaderBundleTimer := metrics.GetOrRegisterTimer("KeyBundleCache.GetTLFReaderKeyBundle", r)
+	putReaderBundleTimer := metrics.GetOrRegisterTimer("KeyBundleCache.PutTLFReaderKeyBundle", r)
+	getWriterBundleTimer := metrics.GetOrRegisterTimer("KeyBundleCache.GetTLFWriterKeyBundle", r)
+	putWriterBundleTimer := metrics.GetOrRegisterTimer("KeyBundleCache.PutTLFWriterKeyBundle", r)
+	// TODO: Implement RatioGauge (
+	// http://metrics.dropwizard.io/3.1.0/manual/core/#ratio-gauges
+	// ) so we can actually display a hit ratio.
+	hitReaderBundleCountMeter := metrics.GetOrRegisterMeter("KeyBundleCache.TLFReaderKeyBundleHitCount", r)
+	hitWriterBundleCountMeter := metrics.GetOrRegisterMeter("KeyBundleCache.TLFWriterKeyBundleHitCount", r)
+	return KeyBundleCacheMeasured{
+		delegate:                  delegate,
+		getReaderBundleTimer:      getReaderBundleTimer,
+		getWriterBundleTimer:      getWriterBundleTimer,
+		putReaderBundleTimer:      putReaderBundleTimer,
+		putWriterBundleTimer:      putWriterBundleTimer,
+		hitReaderBundleCountMeter: hitReaderBundleCountMeter,
+		hitWriterBundleCountMeter: hitWriterBundleCountMeter,
+	}
+}
+
+// GetTLFReaderKeyBundle implements the KeyBundleCache interface for
+// KeyBundleCacheMeasured.
+func (b KeyBundleCacheMeasured) GetTLFReaderKeyBundle(
+	tlfID TlfID, bundleID TLFReaderKeyBundleID) (rkb *TLFReaderKeyBundleV3) {
+	b.getReaderBundleTimer.Time(func() {
+		rkb = b.delegate.GetTLFReaderKeyBundle(tlfID, bundleID)
+	})
+	if rkb != nil {
+		b.hitReaderBundleCountMeter.Mark(1)
```

Comment:
is there a value to counting hits if we aren't also counting total number of attempts?